src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/CacheHealthDecider.java

```diff
@@ -130,7 +146,18 @@ private void configureModifyCacheActionPriority() {
    * signals going forward.
    */
   private Action computeBestAction(final NodeKey esNode, final ResourceEnum cacheType) {
-    return getAction(ModifyCacheMaxSizeAction.NAME, esNode, cacheType, true);
+    final NodeConfigCache nodeConfigCache = getAppContext().getNodeConfigCache();
+    double heapUsedPercent = HeapUtil.getHeapUsedPercentage(
+            NodeConfigCacheReaderUtil.readHeapMaxSizeInBytes(nodeConfigCache, esNode),
+            NodeConfigCacheReaderUtil.readHeapUsageInBytes(nodeConfigCache, esNode));
+
+    // there is no difference in the action when state is UNDER_SIZED or HEALTHY_WITH_BUFFER.
+    // TODO: Tune action parameters based on the state.
+    if (heapUsageBucketCalculator.compute(heapUsedPercent).equals(UsageBucket.HEALTHY_WITH_BUFFER)
+                || heapUsageBucketCalculator.compute(heapUsedPercent).equals(UsageBucket.UNDER_UTILIZED)) {
+      return getAction(ModifyCacheMaxSizeAction.NAME, esNode, cacheType, true);
+    }
+    return null;
```

Comment:
Let us add an info log and add operational metrics for when the action should've been emitted but was punted due to heap utilization.